[Inductor][Tritonparse] Get Inductor kernel params

NikhilAPatel opened the pull request:
Summary: Save the config args that Inductor burns into `inductor_metadata` so we can optionally pass them to any Jit Hooks that are set. This allows us to pass them to Tritonparse.

Reviewed By: davidberard98, FindHao

Differential Revision: D80994791




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

FindHao reviewed (approved):
Thanks for your help!

NikhilAPatel committed:
[Inductor][Tritonparse] Get Inductor kernel params (#161953)

Summary:
X-link: https://github.com/meta-pytorch/tritonparse/pull/89

Pull Request resolved: https://github.com/pytorch/pytorch/pull/161953

Save the config args that Inductor burns into `inductor_metadata` so we can optionally pass them to any Jit Hooks that are set. This allows us to pass them to Tritonparse.

Reviewed By: davidberard98, FindHao

Differential Revision: D80994791
